### PR TITLE
[HLS-3246] - add s3 support + tests

### PR DESCRIPTION
**Pull request by LironKS**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -51,6 +51,13 @@ jobs:
           - name: Checkout Repo
             uses: actions/checkout@v3
 
+          - name: Configure AWS credentials
```
Comment (nozik) [inline .github/workflows/test.yml:55]: Why do we need this?

File: proxy-libs/helioss3/helioss3.go [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+
+type PutObjectTaggingOutput = origin_s3.PutObjectTaggingOutput
+
+const ServiceID = origin_s3.ServiceID
+
+const ServiceAPIVersion = origin_s3.ServiceAPIVersion
+
+type Client = origin_s3.Client
+
+func New(options Options,optFns ...func(*Options) ) (*Client) {
```
Comment (nozik) [inline proxy-libs/helioss3/helioss3.go:30]: Don't we want to patch this as well?

File: proxy-libs/helioss3/helioss3_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+	fmt.Println("Buckets:")
+	for _, bucket := range result.Buckets {
+		fmt.Println(*bucket.Name + ": " + bucket.CreationDate.Format("2006-01-02 15:04:05 Monday"))
+	}
+
+	attributes := assertSpan(t, spanRecorder)
+	fmt.Println(attributes)
+	assertAttributes(t, attributes)
+}
```
Comment (nozik) [inline proxy-libs/helioss3/helioss3_test.go:74]: add newline

File: proxy-libs/helioss3/helioss3_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	// init aws config
+	ctx := context.Background()
+	cfg, err := awsConfig.LoadDefaultConfig(ctx)
+	if err != nil {
+		panic("configuration error, " + err.Error())
+	}
+	s3Client := NewFromConfig(cfg)
+	input := &ListBucketsInput{}
+	result, err := s3Client.ListBuckets(ctx, input)
+	if err != nil {
```
Comment (nozik) [inline proxy-libs/helioss3/helioss3_test.go:61]: err shouldn't be nil, it should fail the test

File: proxy-libs/helioss3/helioss3_test.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		fmt.Printf("Got an error retrieving buckets, %v", err)
+		return
+	}
+
+	fmt.Println("Buckets:")
```
Comment (nozik) [inline proxy-libs/helioss3/helioss3_test.go:66]: remove prints

File: proxy-libs/interface-validator/interface_validator_test.go [hunk 6]
```diff
@@ -70,6 +71,26 @@ func TestEchoInterfaceMatch(t *testing.T) {
 	assert.EqualValues(t, originalExports, heliosExports)
 }
 
+func dropExtractedObject(extractedObjects []exportsExtractor.ExtractedObject, objectNames []string) []exportsExtractor.ExtractedObject {
```
Comment (nozik) [inline proxy-libs/interface-validator/interface_validator_test.go:74]: Why not use `deleteByName`?

**Review by nozik — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (LironKS) [reply .github/workflows/test.yml:55]: for s3 test

Comment (nozik) [reply .github/workflows/test.yml:55]: We shouldn't use prod credentials for unit tests. Use localstack instead

**Review by nozik — APPROVED** [review]

**Pull request merged**